sw/device/lib/dif/dif_pinmux.c

```diff
@@ -5,19 +5,187 @@
 #include "sw/device/lib/dif/dif_pinmux.h"
 
 #include "sw/device/lib/base/bitfield.h"
+#include "sw/device/lib/base/mmio.h"
 #include "sw/device/lib/dif/dif_base.h"
 
 #include "pinmux_regs.h"  // Generated.
 
-OT_WARN_UNUSED_RESULT
+static bool dif_pinmux_get_reg_offset(dif_pinmux_index_t index,
+                                      dif_pinmux_pad_kind_t kind,
+                                      ptrdiff_t mio_reg_base,
+                                      ptrdiff_t dio_reg_base,
+                                      ptrdiff_t *reg_offset) {
+  uint32_t num_pads;
+  ptrdiff_t reg_base;
+
+  switch (kind) {
+    case kDifPinmuxPadKindMio:
+      num_pads = PINMUX_PARAM_N_MIO_PADS;
+      reg_base = mio_reg_base;
+      break;
+    case kDifPinmuxPadKindDio:
+      num_pads = PINMUX_PARAM_N_DIO_PADS;
+      reg_base = dio_reg_base;
+      break;
+    default:
+      return false;
+  }
+  if (index >= num_pads) {
+    return false;
+  }
+  *reg_offset = reg_base + index * sizeof(uint32_t);
+  return true;
+}
+
+static bool dif_pinmux_get_sleep_status_bit(dif_pinmux_pad_kind_t kind,
+                                            dif_pinmux_index_t index,
+                                            ptrdiff_t *reg_offset,
+                                            bitfield_bit32_index_t *bit) {
+  uint32_t num_pads;
+  ptrdiff_t reg_base;
+
+  switch (kind) {
+    case kDifPinmuxPadKindMio:
+      num_pads = PINMUX_PARAM_N_MIO_PADS;
+      reg_base = PINMUX_MIO_PAD_SLEEP_STATUS_0_REG_OFFSET;
+      break;
+    case kDifPinmuxPadKindDio:
+      num_pads = PINMUX_PARAM_N_DIO_PADS;
+      reg_base = PINMUX_DIO_PAD_SLEEP_STATUS_REG_OFFSET;
```

Comment:
It seems unfortunate that this macro's name can change based on the value of a parameter. Perhaps multireg types should _always_ include 0 in the name of the first register, without regard for whether there were enough bits to spill over to a second register.

Reply:
@tjaychen @msfschaffner

Reply:
Huh, this seems to be an old issue that never got properly fixed: https://github.com/lowRISC/opentitan/issues/2254.
We should go ahead and just make the fix (I am checking on that issue what the status is).